.github/actions/build-setup/action.yml

```diff
@@ -1,36 +1,51 @@
 name: "CI Environment Setup"
-description: "Setup a common environment for CI jobs"
-
+description: |
+  Setup a common environment for CI jobs.
+  This should in general be a no-op as it should be run inside a docker container which already has all the components installed.
+  Importantly the installation script checks the versions of tools and will install newer versions if specified for most tools.
 runs:
   using: "composite"
   steps:
     - name: CPUInfo
       shell: bash
       run: |
-        cat /proc/cpuinfo
+        cat /proc/cpuinfo | grep "model name" | head -1
+        cat /proc/cpuinfo | grep "cpu cores" | head -1
     - name: MemInfo
       shell: bash
       run: |
-        cat /proc/meminfo
-    - name: install common dependencies
+        cat /proc/meminfo | grep MemTotal
```

Comment:
Would suggest dumping /proc/meminfo in ./.github/actions/build-teardown so that we can see mem usage delta during the job.

Reply:
It think it's easier in the build setup,  If you suspect the machines are wrong you wouldn't want to wait for the job to fail and be torn down.  Thoughts?

Reply:
Oh, not that the machine would be wrong or different - that is going to be a huge problem.

Here's what I meant by getting the mem delta in the CI job:
1. dump /proc/meminfo at job startup time
2. install deps, compile, run tests, etc
3. dump /proc/meminfo at job teardown time

/proc/meminfo has lots of info on mem usage including swap, which could be helpful to triage/debug the job (such as mem bound).

Reply:
Hmm. There's value there.   Seems we would want to be able to get stats at any point in the build to help debug though.  I'm thinking a separate action/separate pr.

Reply:
Actually this is metrics gathering that should be built in to x and pushed to another service.   Ideally with the test info.